NGitLab/Impl/FilesClient.cs

```diff
@@ -55,6 +70,25 @@ public bool FileExists(string filePath, string @ref)
         }
     }
 
+    public async Task<bool> FileExistsAsync(string filePath, string @ref, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await _api.Head().ExecuteAsync(_repoPath + $"/files/{EncodeFilePath(filePath)}?ref={@ref}", cancellationToken).ConfigureAwait(false);
+            return true;
+        }
+        catch (Exception ex) when (ex is OperationCanceledException ||
+                                  (ex is GitLabException gitLabException && gitLabException.StatusCode == HttpStatusCode.NotFound))
+
+        {
+            return false;
+        }
+        catch (Exception)
+        {
+            throw;
+        }
```

Comment:
I think you can do the same as in the synchronous method:

```suggestion
        catch (GitLabException e) when (e.StatusCode == HttpStatusCode.NotFound)
        {
            return false;
        }
```

Reply:
That's fair. Updated! Let me know.